itou/employee_record/management/commands/transfer_employee_records.py

```diff
@@ -0,0 +1,282 @@
+                continue
+
+            if not dry_run:
+                employee_record.rejected_by_asp(processing_code, processing_label)
+            else:
+                self.logger.info(
+                    "DRY-RUN: Rejected %s, code: %s, label: %s", employee_record, processing_code, processing_label
+                )
+
+        return record_errors
+
+    def download(self, conn, dry_run):
+        """
+        Fetch remote ASP file containing the results of the processing
+        of a batch of employee records
+        """
+        self.logger.info("Starting DOWNLOAD")
+
+        parser = JSONParser()
+        count = 0
+        errors = 0
+
+        # Get into the download folder
+        with conn.cd(settings.ASP_FS_DOWNLOAD_DIR):
+            result_files = conn.listdir()
+
+            if len(result_files) == 0:
+                self.logger.info("No feedback files found")
+                return
+
+            for result_file in result_files:
+                try:
+                    with BytesIO() as result_stream:
+                        self.logger.info("Fetching file '%s'", result_file)
+
+                        conn.getfo(result_file, result_stream)
+                        # Rewind stream
+                        result_stream.seek(0)
+
+                        # Parse and update employee records with feedback
+                        errors += self._parse_feedback_file(result_file, parser.parse(result_stream), dry_run)
+
+                        count += 1
+                except Exception as ex:
+                    errors += 1
+                    self.logger.error("Error while parsing file '%s': %s", result_file, ex)
+
+                self.logger.info("Parsed %s/%s files", count, len(result_files))
+
+            # There were errors do not delete file
+            if errors > 0:
+                self.logger.warning(
+                    "Did not delete file '%s' because of errors. Leaving it in place for another pass...", result_file
+                )
+                return
+
+            # All employee records processed, we can delete feedback file from server
+            if dry_run:
+                self.logger.info("DRY-RUN: Removing file '%s'", result_file)
+                return
+
+            self.logger.info("Deleting '%s' from SFTP server", result_file)
+
+            conn.remove(result_file)
+
+    def upload(self, sftp, dry_run):
+        """
+        Upload a file composed of all ready employee records
+        """
+        self.logger.info("Starting UPLOAD")
+
+        for batch in chunks(EmployeeRecord.objects.ready(), EmployeeRecordBatch.MAX_EMPLOYEE_RECORDS):
+            self._upload_batch_file(sftp, batch, dry_run)
+
+    def handle(self, upload=True, download=True, verbosity=1, dry_run=False, **options):
+        """
+        Employee Record Management Command
+        """
+        if verbosity > 1:
+            self.logger.setLevel(logging.DEBUG)
+
+        both = not (download or upload)
```

Comment:
Cette ligne est difficile à capter, pour moi `both = download and upload`.

Reply:
J'ai l'impression qu'on a 3 cas :

cas 1 - option upload à True. On fait que l'upload.
cas 2 - option download à True. On fait que le download.
cas 3 - aucune des deux options. On fait tout (upload + download).

Le cas 3 est très confusant car upload et download sont à False, et pourtant on fait les 2 actions.

Je propose ceci :

```
if download is False and upload is False:
    # By default we do both upload and download.
    upload = True
    download = True
```

Ensuite le code  plus bas devient plus lisible. Plus besoin de `both`.

Reply:
Je vois que j'ai noyé tous le monde avec `both`.
Même moi au final. 
C'est inutile finalement, c'est le cycle complet par défaut et `download` ou `upload` plus spécifiquement.
Je modifie.